python/doc/theory/data_analysis/smirnov_test.rst

```diff
@@ -1,52 +1,63 @@
-This distance is expressed as follows:
+Two-sample Kolmogorov-Smirnov test
+----------------------------------
+
+The Kolmogorov-Smirnov test is a statistical test that can be used
+to test whether two given samples of data are drawn from the same
+distribution which is of dimension 1 and continuous.
+
+We denote by :math:`\left\{ x_1,\ldots,x_{\sampleSize} \right\}`
+and :math:`\left\{ y_1,\ldots,y_{\sampleSize} \right\}` two given samples of dimension 1.
+Let :math:`F` be  the (unknown) cumulative distribution function of
+:math:`\left\{ x_1,\ldots,x_{\sampleSize} \right\}` and :math:`G` the one
+of :math:`\left\{ y_1,\ldots,y_{\sampleSize} \right\}`.
+
+We want to test  whether both samples are drawn from the same distribution, ie whether :math:`F=G`.
+
+
+This test involves the calculation of the test statistic which is the weighted maximum
+distance between both empirical cumulative distribution functions
+:math:`\widehat{F}_{X}` and :math:`\widehat{F}_{Y}` built respectively from
+:math:`\left\{ x_1,\ldots,x_{\sampleSize} \right\}`
+and :math:`\left\{ y_1,\ldots,y_{\sampleSize} \right\}`.
+Letting :math:`X_1, \ldots , X_\sampleSize` be i.i.d. random variables following the distribution
+with CDF :math:`F`, and :math:`Y_1, \ldots , Y_\sampleSize` be i.i.d. random variables following
+the distribution with CDF :math:`G`, the test statistic is defined by:
+
+.. math::
+
+    D_{\sampleSize} = \sqrt{\sampleSize} \sup_x \left|\widehat{F}_{X}\left(x\right) - \widehat{F}_{Y}\left(x\right)\right|
+
+The empirical value of the test statistic, evaluated from the sample is:
 
 .. math::
 
-   \begin{aligned}
-       \widehat{D}_{M,N} = \sup_x \left|\widehat{F}_N\left(x\right) - \widehat{F}'_M\left(x\right)\right|
-     \end{aligned}
-
-The probability distribution of the distance :math:`\widehat{D}_{M,N}`
-is asymptotically known (i.e. as the size of the samples tends to
-infinity). If :math:`M` and :math:`N` are sufficiently large, this means
-that for a probability :math:`\alpha`, one can calculate the threshold /
-critical value :math:`d_\alpha` such that:
-
--  if :math:`\widehat{D}_{M,N} >d_{\alpha}`, we conclude that the two
-   samples are not identically distributed, with a risk of error
-   :math:`\alpha`,
-
--  if :math:`\widehat{D}_{M,N} \leq d_{\alpha}`, it is reasonable to say
-   that both samples arise from the same distribution.
-
-An important notion is the so-called “:math:`p`-value” of the test. This
-quantity is equal to the limit error probability
-:math:`\alpha_\textrm{lim}` under which the “identically-distributed”
-hypothesis is rejected. Thus, the two samples will be supposed
-identically distributed if and only if :math:`\alpha_\textrm{lim}` is
-greater than the value :math:`\alpha` desired by the user. Note that the
-higher :math:`\alpha_\textrm{lim} - \alpha`, the more robust the
-decision.
-
-This test is also referred to as the Kolmogorov-Smirnov’s test for two
-samples.
+    d = \sqrt{\sampleSize} \sup_{x_i}
```

Comment:
The sup should be on the index i, not on x, because otherwise where does $y_i$ come from ?

(This may hold for several files, I am not sure.)

Reply:
ok. The test was badly written... I modified it! see the new version..... The sum is not indexed by the values in the sample!!!!